site/docs/v1/tech/apis/identity-providers/_identity-provider-login-request-body.adoc

```diff
@@ -48,9 +48,15 @@ ifeval::["{loginProvider}" == "Facebook"]
 [field]#data.token# [type]#[String]# [required]#Required#::
 The Facebook access token returned from their login API. This token will be sent to the Facebook Me Graph API as the `access_token` parameter.
 endif::[]
-ifeval::["{loginProvider}" == "Google"]
+ifeval::["{idp_type}" == "Google"]
+[field]#data.code# [type]#[String]# [required]#Required#::
+The Google `code` when not using `token` returned from their login API. This token will be sent to the Google OAuth Token Info API as the `id_token` parameter.  When using `code` you must also supply `redirect_uri`.
+
+[field]#data.redirect_uri# [type]#[String]# [required]#Required#::
+The redirect URI that was provided to the Google Authorization endpoint. This value will be sent to the Token Info API as the `redirect_uri` parameter.
+
 [field]#data.token# [type]#[String]# [required]#Required#::
-The Google id token returned from their login API. This token will be sent to the Google OAuth Token Info API as the `id_token` parameter.
+The Google id token when not using `code` returned from their login API. This token will be sent to the Google OAuth Token Info API as the `id_token` parameter.  When using `token` you do not have to supply `redirect_uri`.
 endif::[]
```

Comment:
Note that I left all three of these as **Required**, but their required-ness is that you must supply one or the other.  More specifically, you must submit `code` and `redirect_uri` or `token`.  Let me know if there is a better way to make this clear.

Reply:
Hmmm. We should standardize on this. I think in other areas we have done the opposite. Here: https://fusionauth.io/docs/v1/tech/apis/users/

If you search for username or email, they are both marked as optional, then in the text description, it specifies one is required: ` If email is not provided, then the username will be required.`

That's what I'd vote for here. (And we should add this to the DocsDevReadme as well--can you please do that?)

Reply:
I did some searching and found that in education they use the term "Conditional" or "Conditionally Required".  Not sure how the `[required]` prefix works in AsciiDoc.  Does it just apply styling?  We could use the color for required and the term "Conditional", `[required]#Conditional#`, that way it would stand out to the user that there is more going on here than just an optional field.  People tend to skim over the plain text and stop on the red and orange text.  Especially, when you're trying to figure out why it's not working.